### DISCUSSION TITLE
SD MP3 player metadata

### DISCUSSION THREAD

**Discussion by freakstreet** [pschatzmann/arduino-audio-tools#1842] [Q&A] [answered]
Hi,
I'm sorry for this dumb question. I am testing an esp32 to read and output audio from mp3 files on SD card. All your developped libraries are incredible to help people like me.

MP3 metadata seems to interfere with audio reading. I reworked my files removing meta, reading is perfect.

I've seen a thread about a metadata filter but didn't understand how to implement it on player.

_MetaDataFilter filter(decoder);
**and copy the data to the filter**._

Actually I'm having in the loop:
player.copy()

Would you guide me please ?
Best

Laurent

**Comment by pschatzmann** [comment, marked as answer]
For the player you can use the [MetaDataFilterDecoder](https://pschatzmann.github.io/arduino-audio-tools/classaudio__tools_1_1_meta_data_filter_decoder.html):

```C++
I2SStream i2s;
MP3DecoderHelix mp3;  
MetaDataFilterDecoder filter(mp3);
AudioPlayer player(source, i2s, filter);
```

**Reply by freakstreet** [reply]
Thank you very much for this fast answer.
Of course the code worked as you suggested but using the filter my app was crashing, even reading mp3 without cover in metadata.
Right now I removed the filter and also removed cover metadata in my mp3 files, this works greatly.

**Reply by pschatzmann** [reply]
Can you provide an example file to reproduce your issue!
